src/coreclr/nativeaot/Common/src/Internal/Runtime/MethodTable.cs

```diff
@@ -1277,13 +1277,17 @@ public uint GetFieldOffset(EETypeField eField)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public ref T GetField<T>(EETypeField eField)
         {
-            return ref Unsafe.As<byte, T>(ref *((byte*)Unsafe.AsPointer(ref this) + GetFieldOffset(eField)));
+#pragma warning disable CS9084 // Struct member returns 'this' or other instance members by reference
+            return ref Unsafe.As<byte, T>(ref Unsafe.AddByteOffset(ref Unsafe.As<MethodTable, byte>(ref Unsafe.AsRef(in this)), (nint)GetFieldOffset(eField)));
+#pragma warning restore CS9084
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public ref T GetField<T>(uint offset)
         {
-            return ref Unsafe.As<byte, T>(ref *((byte*)Unsafe.AsPointer(ref this) + offset));
+#pragma warning disable CS9084 // Struct member returns 'this' or other instance members by reference
```

Comment:
All this ceremony to avoid Unsafe.AsPointer in super low-level code does not look like an improvement. The current code with Unsafe.AsPointer is much more readable than this.